### PR TITLE
Add Some Subscription Tests

### PR DESCRIPTION
**Pull request by jamiebrynes7**
#### Description
Some tests for: 
- the standard `SubscriptionManager` implementations.
- `AggregateSubscription` behaviour
- `RequiredSubscriptionFieldInfo` behaviour

More tests will come in the form of integration tests with Reader/Writer injection. 

#### Tests
Yes we are tests.

#### Documentation
How is this documented (for example: release note, upgrade guide, feature page, in-code documentation)?

**Did you remember a changelog entry?**
N/A

#### Primary reviewers
If your change will take a long time to review, you can name at most two primary reviewers who are ultimately responsible for reviewing this request. @ mention them.

### CODE REVIEW

File: workers/unity/Packages/com.improbable.gdk.core/Tests/Editmode/Subscriptions/StandardSubscriptionTests.cs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    /// <summary>
+    ///     This tests the availability constraints for the following subscriptions :
+    ///         - EntityId
+    ///         - ECS Entity
+    ///         - Log Dispatcher
+    ///         - World Commands
+    ///         - ECS World
+    /// </summary>
+    [TestFixture]
+    public class StandardSubscriptionTests : SubscriptionTestBase
```
Comment (samiwh) [inline workers/unity/Packages/com.improbable.gdk.core/Tests/Editmode/Subscriptions/StandardSubscriptionTests.cs]: What does standard mean

Comment (jamiebrynes7) [reply workers/unity/Packages/com.improbable.gdk.core/Tests/Editmode/Subscriptions/StandardSubscriptionTests.cs]: Its what you called the folder `StandardSubscriptionManagers`

Comment (samiwh) [reply workers/unity/Packages/com.improbable.gdk.core/Tests/Editmode/Subscriptions/StandardSubscriptionTests.cs]: Oh I forgot I used that word.

File: workers/unity/Packages/com.improbable.gdk.core/Tests/Editmode/Subscriptions/RequiredSubscriptionsInfoTests.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        }
+
+        private class RequiredStaticFields
+        {
+            [Require] public static World World;
+            [Require] private static Entity entity;
+        }
+
+        [Test]
+        public void RequiredSubscriptionsInfo_ignores_static_fields()
```
Comment (samiwh) [inline workers/unity/Packages/com.improbable.gdk.core/Tests/Editmode/Subscriptions/RequiredSubscriptionsInfoTests.cs]: We might want to make this error. We could also support them, although that seems dangerous.

Comment (jamiebrynes7) [reply workers/unity/Packages/com.improbable.gdk.core/Tests/Editmode/Subscriptions/RequiredSubscriptionsInfoTests.cs]: Yeah I'd argue for this to be an error as well. A `static` Require feels incorrect

Comment (jamiebrynes7) [reply workers/unity/Packages/com.improbable.gdk.core/Tests/Editmode/Subscriptions/RequiredSubscriptionsInfoTests.cs]: Will open a ticket

Comment (jamiebrynes7) [reply workers/unity/Packages/com.improbable.gdk.core/Tests/Editmode/Subscriptions/RequiredSubscriptionsInfoTests.cs]: Raised UTY-1815

**Pull request merged**